### DISCUSSION TITLE
Adding schema manually to describe the data

### DISCUSSION THREAD

**Discussion by RoyKulik** [sinaptik-ai/pandas-ai#505] [Q&A]
Assuming I want to manually create description of the columns instead of letting the LLM infer it by using df.head() or similar, how can it be done?
I saw this code snippet on another discussion about the code generation instructions:

```
# Creating SmartDataFrames
sdf_sales_continent = SmartDataframe(
    df,
    name="df",
    config={
        "custom_prompts": {
            "generate_python_code": CustomAnalysisPrompt(),
        }
    }
)
```

As far as I understand, it could help me improve performance as I can use a more detailed and most accurate description of the data.

Would appreciate any input.

**Comment by RoyKulik** [comment]
I dived deeper into the code and I see that the place where the description is built is here:

```
    def _generate_dataframes(self, dfs):
        """
        Generate the dataframes metadata
        Args:
            dfs: List of Dataframes
        """
        dataframes = []
        for index, df in enumerate(dfs, start=1):
            description = "Dataframe "
            if df.name is not None:
                description += f"{df.name} (dfs[{index-1}])"
            else:
                description += f"dfs[{index-1}]"
            description += (
                f", with {df.rows_count} rows and {df.columns_count} columns."
            )
            if df.description is not None:
                description += f"\nDescription: {df.description}"
            description += f"""
This is the metadata of the dataframe dfs[{index-1}]:
{df.head_csv}"""  # noqa: E501
            dataframes.append(description)

        return "\n\n".join(dataframes)
```

So I guess using the description argument can work 🥇 .
I didn't understand it from the documentation so maybe adding an example in the examples can be useful.

My case suits this case where I want to enforce_privacy and still be able to solve complex queries by providing detailed description.

**Reply by octadion** [reply]
I also want to know how, I want to include a description in the metadata, have you been able to do it? Thank You
